coderd/templates.go

```diff
@@ -855,6 +861,35 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, api.convertTemplate(updated))
 }
 
+func (api *API) notifyUsersOfTemplateDeprecation(ctx context.Context, template database.Template) error {
+	users, err := api.Database.GetUsersWithAccessToTemplateByID(ctx, template.ID)
+	if err != nil {
+		return xerrors.Errorf("get users with access to template by id: %w", err)
+	}
+
+	errs := []error{}
+
+	for _, userID := range users {
+		_, err = api.NotificationsEnqueuer.Enqueue(
+			//nolint:gocritic // We need the system auth context to be able to send the deprecation notification.
+			dbauthz.AsSystemRestricted(ctx),
```

Comment:
Let's add a follow-up issue to add a separate RBAC role specifically for enqueuing notifications, if it doesn't exist already.